src/Php80/Php80.php

```diff
@@ -83,4 +83,14 @@ public static function str_contains(string $haystack, string $needle): bool
     {
         return '' === $needle || false !== strpos($haystack, $needle);
     }
+
+    public static function str_starts_with(string $haystack, string $needle): bool
+    {
+        return 0 === \strncmp($haystack, $needle, \strlen($needle));
+    }
+
+    public static function str_ends_with(string $haystack, string $needle): bool
+    {
+        return '' === $needle || $needle === \substr($haystack, -\strlen($needle));
```

Comment:
Using `substr()` is suboptimal (as [pointed in the RFC](https://wiki.php.net/rfc/add_str_starts_with_and_ends_with_functions#str_ends_with)). The [actual implementation](https://github.com/php/php-src/pull/5300/files?file-filters%5B%5D=.c) rather does something akin to `substr_compare()`, which could give:

```php
        return '' === $needle || ('' !== $haystack && 0 === \substr_compare($haystack, $needle, -\strlen($needle)));
```

(see https://3v4l.org/NUS87 and the "based on" links for why the two guards)

Reply:
Please provide a PR and some benchmark if possible.

Reply:
See #255